### PR TITLE
NoBug- Convert tests with local webserver to EarlGrey

### PR DESCRIPTION
**Pull request by npark-mozilla**
Since simplewebserver is not accessible in UI Test environment. This also includes tests that inject values into internal variables.
6 more tests in UITests folder are added to the FirefoxNightly for regualr testing.
- BrowserTests
- HistoryTests
- LoginInputTests
- ReadingListTests
- SecurityTests
- SessionRestoreTests

### CODE REVIEW

**Comment by npark-mozilla** [comment]
@farhanpatel r?

**Comment by farhanpatel** [comment]
### SwiftLint found issues

#### Warnings

File | Line | Reason |
| --- | ----- | ----- |
LoginInputTests.swift | 50 | Else and catch should be on the same line, one space after the previous declaration. 
#### Errors

File | Line | Reason |
| --- | ----- | ----- |
LoginInputTests.swift | 105 | There should be no space before and one after any comma. 
LoginInputTests.swift | 125 | There should be no space before and one after any comma. 


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: UITests/SessionRestoreTests.swift [hunk 1]
```diff
@@ -34,25 +34,44 @@ class SessionRestoreTests: KIFTestCase {
...
             canGoBack = true
         } catch _ {
             canGoBack = false
         }
         XCTAssertFalse(canGoBack, "Reached the beginning of browser history")
-        tester().tapViewWithAccessibilityLabel("Forward")
-        tester().tapViewWithAccessibilityLabel("Forward")
-        tester().tapViewWithAccessibilityLabel("Forward")
-        tester().waitForWebViewElementWithAccessibilityLabel("Page 3")
+
```
Comment (farhanpatel) [inline UITests/SessionRestoreTests.swift]: Maybe its just github. But the indentation looks wrong here.

Comment (npark-mozilla) [inline UITests/SessionRestoreTests.swift]: ah, i think it's probably my xcode using tab instead of space for indentation, I'll fix this.

**Comment by npark-mozilla** [comment]
I fixed the syntax for swift 3, and since it's r+ed, I'll merge them now against the swift 3 migration branch.

**Pull request merged**